src/library/scala/collection/StringOps.scala

```diff
@@ -492,39 +494,52 @@ final class StringOps(private val s: String) extends AnyVal {
     *               the string `sep`.
     */
   def mkString(start: String, sep: String, end: String): String =
-    addString(new mutable.StringBuilder(), start, sep, end).toString
+    addString(new jStringBuilder(), start, sep, end).toString
 
   /** Displays all elements of this string in a string using a separator string.
     *
     *  @param sep   the separator string.
     *  @return      In the resulting string
     *               the chars of this string are separated by the string `sep`.
     */
-  def mkString(sep: String): String = mkString("", sep, "")
+  def mkString(sep: String): String =
+    if (sep.length == 0) s
+    else mkString("", sep, "")
 
   /** Returns this String */
   def mkString: String = s
 
   /** Appends this string to a string builder. */
-  def addString(b: mutable.StringBuilder): mutable.StringBuilder =
+  def addString(b: StringBuilder): StringBuilder =
     b.append(s)
 
   /** Appends this string to a string builder using a separator string. */
-  def addString(b: mutable.StringBuilder, sep: String): mutable.StringBuilder =
+  def addString(b: StringBuilder, sep: String): StringBuilder =
     addString(b, "", sep, "")
 
   /** Appends this string to a string builder using start, end and separator strings. */
-  def addString(b: mutable.StringBuilder, start: String, sep: String, end: String): mutable.StringBuilder = {
+  def addString(b: StringBuilder, start: String, sep: String, end: String): StringBuilder = {
+    addString(b.sb, start, sep, end)
+    b
+  }
+
+  /** Appends this string to a string builder using start, end and separator strings. */
+  def addString(b: jStringBuilder, start: String, sep: String, end: String): jStringBuilder = {
     b.append(start)
     val len = s.length
-    var i = 0
-    while(i < len) {
-      if(i != 0) b.append(sep)
-      b.append(s.charAt(i))
-      i += 1
+    if (len > 0) {
+      if (sep.length == 0) b.append(s)
+      else {
+        b.append(s.charAt(0))
+        var i = 1
+        while (i < len) {
```

Comment:
Did you benchmark this? From my experience with optimizing `ArrayOps.fold` I expect that this version will be slower. The original one has an extra branching operation inside the loop but it loops from 0 for which the JVM has extra optimizations.

Reply:
I'll run a benchmark.

Reply:
ran a jmh with java 8 and this version is significantly faster